File: Logic_HandsOn/C++/Problems on Digits,String,Array,Pointers etc/program97.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and count number of small characters.				
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////



#include<iostream>
using namespace std;

class Solution
{

	private:
		string str;
		int iCnt,iCount;

	public:

		Solution(string str,int iCnt=0,int iCount=0)
		{
			this->str=str;
		}

		int CountSmall()
		{
			for(iCnt=0;iCnt<str.length();iCnt++)
			{
				if(str[iCnt]>='a' && str[iCnt]<='z')
				{
					iCount++;
				}
			}	
			return iCount;
		}
};

int main()
{
	string str;
	int iRet=0;

	cout<<"Enter string: ";
	getline(cin,str);
	
	Solution *sObj = new Solution(str);
	iRet = sObj->CountSmall();
	cout<<"Small are: "<<iRet<<"\n";

	delete sObj;
	return 0;

}

/*
OUTPUT:

Enter string: Marvellous
Small are: 9

*/

File: Logic_HandsOn/Java/Problems on Array,Pattern etc/Program97.java
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and count number of small characters.				
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Solution
{
	String str;
	int iCnt,iCount;

	Solution(String str)
	{
		this.str=str;
		this.iCnt=0;
		this.iCount=0;
	}

	int CountSmall()
	{
		
		while(iCnt!=str.length())
		{
			if(str.charAt(iCnt)>='a' && str.charAt(iCnt)<='z')
			{
				iCount++;
			}
			iCnt++;
		}
		
		return iCount;
	}
}

class Program
{
	public static void main(String args[])
	{
		int iRet=0;
		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter the string: ");
		String str = sObj.nextLine();

		Solution obj = new Solution(str);
		iRet = obj.CountSmall();

		System.out.println("Small are: "+iRet);
	}
}

/*
OUTPUT:
Enter the string: 
Vishal
Small are: 5

*/
